packages/apps/haaretz.co.il/components/ArchivePage/elements/Calendar.js

```diff
@@ -71,12 +73,21 @@ export default function Calendar({ yearValue, monthValue, setDayValue, dayValue,
     };
   }, [ calendarOpen, ]);
 
+  React.useEffect(() => {
+    if (isMobile) {
+      setCalendarOpen(false);
+    }
+    // eslint-disable-next-line
+  }, [dayValue]);
```

Comment:
```suggestion
  }, [ dayValue, isMobile, ]);
```
You depend on `isMobile` here, so you need it in the dependencies array. Otherwise you might have a reference to the wrong value because of a stale closure.